app/services/onlinetesting/ApplicationStatusCalculator.scala

```diff
@@ -27,20 +28,43 @@ import model.persisted.SchemeEvaluationResult
 
 trait ApplicationStatusCalculator {
 
-  def determineApplicationStatus(originalApplicationStatus: ApplicationStatus,
-                                 evaluatedSchemes: List[SchemeEvaluationResult],
-                                 phase: Phase): Option[ProgressStatus] = {
+  def determineApplicationStatus(applicationRoute: ApplicationRoute,
+    originalApplicationStatus: ApplicationStatus,
+    evaluatedSchemes: List[SchemeEvaluationResult],
+    phase: Phase): Option[ProgressStatus] = {
     val results = evaluatedSchemes.map(s => Result(s.result))
     require(results.nonEmpty, "Results not found")
-    (phase, originalApplicationStatus) match {
+
+    def faststreamStatusCalculator = (phase, originalApplicationStatus) match {
       case (PHASE1, ApplicationStatus.PHASE1_TESTS) => processResults(results, PHASE1_TESTS_PASSED, PHASE1_TESTS_FAILED)
+
       case (PHASE2, ApplicationStatus.PHASE2_TESTS) => processResults(results, PHASE2_TESTS_PASSED, PHASE2_TESTS_FAILED)
+
       case (PHASE3, ApplicationStatus.PHASE3_TESTS | ApplicationStatus.PHASE3_TESTS_PASSED_WITH_AMBER)
         if results.contains(Amber) && results.contains(Green) => Some(PHASE3_TESTS_PASSED_WITH_AMBER)
+
       case (PHASE3, ApplicationStatus.PHASE3_TESTS | ApplicationStatus.PHASE3_TESTS_PASSED_WITH_AMBER) =>
         processResults(results, PHASE3_TESTS_PASSED, PHASE3_TESTS_FAILED)
+
+      case _ => None
+    }
+
+    def edipStatusCalculator = (phase, originalApplicationStatus) match {
+      case (PHASE1, ApplicationStatus.PHASE1_TESTS | ApplicationStatus.PHASE1_TESTS_PASSED_WITH_AMBER)
+        if results.contains(Amber) && results.contains(Green) => Some(PHASE1_TESTS_PASSED_WITH_AMBER)
+
+      case (PHASE1, ApplicationStatus.PHASE1_TESTS | ApplicationStatus.PHASE1_TESTS_PASSED_WITH_AMBER) =>
+        processResults(results, PHASE1_TESTS_PASSED, PHASE1_TESTS_FAILED)
+
       case _ => None
     }
+
+
+    applicationRoute match {
+      case ApplicationRoute.Faststream => faststreamStatusCalculator
+
+      case ApplicationRoute.Edip => edipStatusCalculator
```

Comment:
Processing for those applications isn't implemented yet so I thought it would be enough to just propagate the exception up to the scheduler.  This scheduler does batch evaluation and catches the failed futures to a Failure and carries on processing the remaining applications in the batch.